resources/netlog_defs.ts

```diff
@@ -812,24 +814,36 @@ const logDefinitions = {
   },
 } as const;
 
+export const logDefinitions = {
+  en: intlLogDefinitions,
+  de: intlLogDefinitions,
+  fr: intlLogDefinitions,
+  ja: intlLogDefinitions,
+  cn: intlLogDefinitions,
+  ko: intlLogDefinitions,
+};
+
 // Verify that this has the right type, but export `as const`.
 const assertLogDefinitions: LogDefinitionMap = logDefinitions;
 console.assert(assertLogDefinitions);
 
 export type LogDefinitions = typeof logDefinitions;
-export type LogDefinitionTypes = keyof LogDefinitions;
+export type LogDefinitionTypes = {
+  [lang in Lang]: keyof LogDefinitions[lang];
+};
 
 export type ParseHelperField<
-  Type extends LogDefinitionTypes,
-  Fields extends NetFieldsReverse[Type],
+  Type extends LogDefinitionTypes[L],
+  Fields extends NetFieldsReverse<L>[Type],
   Field extends keyof Fields,
+  L extends Lang = 'en',
 > = {
   field: Fields[Field] extends string ? Fields[Field] : never;
   value?: string;
 };
 
-export type ParseHelperFields<T extends LogDefinitionTypes> = {
-  [field in keyof NetFieldsReverse[T]]: ParseHelperField<T, NetFieldsReverse[T], field>;
+export type ParseHelperFields<T extends LogDefinitionTypes[L], L extends Lang = 'en'> = {
```

Comment:
(I don't think I've seen a template like this with extends and an equal.  Is this a default template parameter?)

Reply:
Yes, this has been supported since TS 2.3, as far as I can see, it's just non-existent in their actual documentation:

https://github.com/Microsoft/TypeScript/wiki/Roadmap#23-april-2017

It's generally not a good idea for a number of reasons related to compatibility between definitions, which is why I stated that the goal before merging this PR is to remove the default type specifiers.

An example of bad compatibility, adding a default value to a function parameter for a template property with a default value:
![image](https://user-images.githubusercontent.com/6119598/158252472-d16d51d4-ac0d-4746-9a25-557837999edd.png)